preprocessy/encoding/_categorical.py

```diff
@@ -91,47 +68,78 @@ def __encode_categorical_util(self):
         if demanded.
         """
-        if self.cat_cols is not None:
-            self.__encode_categorical_util()
-        else:
+        if self.cat_cols is None:
             rows = self.train_df.shape[0]
             rows = 0.5 * rows
             self.cat_cols = []
             for col in self.train_df.columns:
-                if (
-                    "$" in self.train_df[col][0]
-                    or self.train_df[col].str.contains(",").any()
-                ):
-                    self.train_df[col] = (
-                        self.train_df[col]
-                        .apply(lambda x: x.replace("$", "").replace(",", ""))
-                        .astype("float")
-                    )
-                elif pd.to_datetime(
-                    self.train_df[col], errors="coerce"
-                ).isnull().sum() < 0.7 * len(self.train_df[col]):
-                    self.train_df[col] = pd.to_datetime(
+                if col not in self.ord_cols:
+                    if (
+                        "$" in self.train_df[col][0]
+                        or self.train_df[col].str.contains(",").any()
+                    ):
+                        self.train_df[col] = (
+                            self.train_df[col]
+                            .apply(lambda x: x.replace("$", "").replace(",", ""))
+                            .astype("float")
+                        )
+                    elif pd.to_datetime(
                         self.train_df[col], errors="coerce"
-                    )
-                elif (
-                    isinstance(self.train_df[col][0], int)
-                    and self.train_df[col].nunique() < rows
-                ):
-                    self.cat_cols.append(col)
-                elif (
-                    isinstance(self.train_df[col][0], str)
-                    and self.train_df[col].nunique() < rows
-                ):
-                    self.cat_cols.append(col)
-            self.__encode_categorical_util()
+                    ).isnull().sum() < 0.7 * len(self.train_df[col]):
+                        self.train_df[col] = pd.to_datetime(
+                            self.train_df[col], errors="coerce"
+                        )
+                    elif (
+                        isinstance(self.train_df[col][0], int)
+                        and self.train_df[col].nunique() < rows
+                    ):
+                        self.cat_cols.append(col)
+                    elif (
+                        isinstance(self.train_df[col][0], str)
+                        and self.train_df[col].nunique() < rows
+                    ):
+                        self.cat_cols.append(col)
+                else:
+                    continue
 
+        if self.one_hot:
+            self.__encode_one_hot_util()
+        else:
+            self.__encode_categorical_util()
         return True
```

Comment:
No need to return True here. An empty return statement will be fine.

Reply:
Okay will change this.